synapse/rest/client/v2_alpha/account.py

```diff
@@ -512,35 +683,35 @@ def on_GET(self, request):
 
     @defer.inlineCallbacks
     def on_POST(self, request):
+        requester = yield self.auth.get_user_by_req(request)
+        user_id = requester.user.to_string()
         body = parse_json_object_from_request(request)
 
         threepid_creds = body.get("threePidCreds") or body.get("three_pid_creds")
         if threepid_creds is None:
             raise SynapseError(
                 400, "Missing param three_pid_creds", Codes.MISSING_PARAM
             )
+        assert_params_in_dict(threepid_creds, ["client_secret", "sid"])
 
-        requester = yield self.auth.get_user_by_req(request)
-        user_id = requester.user.to_string()
-
-        # Specify None as the identity server to retrieve it from the request body instead
-        threepid = yield self.identity_handler.threepid_from_creds(None, threepid_creds)
```

Comment:
should we not continue to delegate to the identity server if `threepid_behaviour_email == ThreepidBehaviour.REMOTE` ? Otherwise the user is going to do the validation via the IS and it's not going to help them?

Reply:
Very good point. There's also the added complication that we don't know the medium of the 3PID that the user's validation session is referring to, so we don't know which account_threepid_delegate to check if we have any configured.

I've changed things so that we first assume it's an email, try remote if set up, then local. And if that doesn't work, assume it's a phone number, and try our configured delegate for that if available.

If nobody states that the provided client_secret and sid have been validated, then we fail.